### PR TITLE
Ban ignored users from watching

### PR DESCRIPTION
**Pull request by aastrand**
This commit bans ignored users from watching
games where the creator has them on their ignore list.

### CODE REVIEW

File: Mage.Client/src/main/java/mage/client/game/GamePanel.java [hunk 1]
```diff
@@ -305,10 +305,14 @@ public void cleanUp() {
...
         this.playersWhoLeft.clear();
 
-        jLayeredPane.remove(abilityPicker);
+        if (jLayeredPane!= null) {
+            jLayeredPane.remove(abilityPicker);
+        }
         this.abilityPicker.cleanUp();
 
-        jLayeredPane.remove(DialogManager.getManager(gameId));
+        if (jLayeredPane != null) {
```
Comment (drmDev) [inline Mage.Client/src/main/java/mage/client/game/GamePanel.java]: minor: couldn't this `jLayeredPane.remove(DialogManager.getManager(gameId));` be moved up to the other conditional checking on `jLayeredPane!= null` on line 308?

Comment (aastrand) [inline Mage.Client/src/main/java/mage/client/game/GamePanel.java]: Seems like it. Will fix.

**Comment by drmDev** [comment]
Difficult one to review due to difficult to test and it is updating existing code that could potentially have unwanted effects. Code seems fine. I'll let others take a closer look at it, offer feedback, before we merge it in.

**Comment by aastrand** [comment]
Agreed it is annoying to test with actual test code.
Did test it manually of course but I could've missed an edge case.
